### PR TITLE
Fix: Validation of fields with a leading asterisk.

### PR DESCRIPTION
**Pull request by iRedds**
**Description**
https://forum.codeigniter.com/showthread.php?tid=82391&pid=398147#pid398147

Fixed: Validation of fields with a leading asterisk (wildcard).
Also fixed getting an error by field name with a leading asterisk.

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [x] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Review by ddevsr — APPROVED** [review]

**Review by MGatner — APPROVED** [review]
I'm not great with regex patterns but your test case looks good and nothing else failed so it's good by me. Thank you!

**Comment by MGatner** [comment]
Original reporter verifies this works 💪

File: tests/system/Validation/ValidationTest.php [hunk 1]
```diff
@@ -1324,4 +1330,17 @@ public function testNestedArrayThrowsException(): void
...
     }
+
+    public function testRuleWithLeadingAsterisk(): void
+    {
+        $data = [
+            ['foo' => 1],
+            ['foo' => null],
+        ];
+
+        $this->validation->setRules(['*.foo' => 'required'], ['1.foo' => ['required' => 'Required {field}']]);
```
Comment (kenjis) [inline tests/system/Validation/ValidationTest.php:1341]: This is not related with this bug, but `['1.foo' => ['required' => 'Required {field}']]` seems tricky .

**Review by kenjis — APPROVED** [review]

**Comment by kenjis** [comment]
Thank you! @iRedds

File: tests/system/Validation/ValidationTest.php (hunk 1 above)
Comment (iRedds) [reply tests/system/Validation/ValidationTest.php:1341]: You know, it turned out that the option you suggested does not work. The default error text is returned.

Comment (iRedds) [reply tests/system/Validation/ValidationTest.php:1341]: The error text will be defined for the ``*.foo`` field, and the ``1.foo`` field is passed.

**Pull request merged**